src/service/resources-impl.js

```diff
@@ -1402,6 +1408,29 @@ export class Resources {
       // Selection API not supported.
     }
   }
+
+  /**
+   * Cleanup task queues from tasks for elements that has been unloaded.
+   * @param resource
+   * @private
+   */
+  maybeRemoveFromQueues_(resource) {
+    if (resource.getState() == ResourceState_.NOT_LAID_OUT) {
+      // If the layout promise for this resource has not resolved yet, remove
+      // it from the task queues to make sure this resource can be rescheduled
+      // for layout again later on.
+      // TODO(mkhatib): Think about how this might affect preload tasks once the
+      // prerender change is in.
+      this.queue_.filter(task => {
+        return task.resource != resource;
+      });
+      this.exec_.filter(task => {
+        return task.resource != resource;
+      });
+      this.requestsChangeSize_ =  this.requestsChangeSize_.filter(
```

Comment:
Let's rename the tasks's API then. Maybe let's call it `purge`? Or `cleanup` or such. But using `filter` is definitely misleading.